streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1248,15 +1248,16 @@ protected StreamTask createStreamTask(final TaskId id, final Collection<TopicPar
     }
 
     private Producer<byte[], byte[]> createProducer(final TaskId id) {
-        final Map<String, Object> producerConfigs = config.getProducerConfigs(threadClientId);
```

Comment:
This makes me think (in a more general sense), that what would be the upgrade path for people to switch from at-least-once to exactly-once? For example in this case all the metrics would be broken as we are not having new clients. I looks to me that operation-wise it would be some burden on users to switch on / off this knob.

Anyways, this may be a more general discussion that we can have separately.